### ISSUE TITLE
Who is who ?

### ISSUE THREAD

**Issue by xakraz** [goabout/kustomize-sopssecretgenerator#9]
Hi !

I was looking for a kustomize compatible secrets managent and fall on 3 projetcs that are very similar:
* Yours
* [monopole/sopsencodedsecrets](https://github.com/monopole/sopsencodedsecrets) which is used in `kustomize` repository as an example
* [viaduct-ai/kustomize-sops](https://github.com/viaduct-ai/kustomize-sops) which is referenced on `ArgoCD` documentation

So I am wondering, are you all aware that you are doing the same effort ?

**Comment by jcassee** [comment]
@xakraz I see you did not create similar issues on the other projects, not sure they know this issue exists... :-D

The main reason d'etre for this project is that is builds an executable, which is less brittle than using the Go plugin interface. With the support for annotations for resource generation options (kubernetes-sigs/kustomize#1473) external plugins can behave exactly like the pre-3.0 secret generators.

Although I'm happy with the way this plugin works, I'm more interested in having _a_ sops secrets generator than it being this one.

**Comment by devstein** [comment]
@jcassee Thanks for the notification 😄 

@xakraz Great question! When we first created [ksops](https://github.com/viaduct-ai/kustomize-sops), this project's exec plugin approach did not exist and @monopole's solution only worked with secrets and we wanted the flexibility to encrypt other resources too. I also added support for [resource generation options annotations](https://github.com/viaduct-ai/kustomize-sops/pull/26) to have parity with exec plugins.


My team was focused on creating a GitOps secret management solution for Argo CD (hence its location in the documentation), but I completely agree with @jcassee comment:

>Although I'm happy with the way this plugin works, I'm more interested in having a sops secrets generator than it being this one.

I think the reason there are so many similar projects is that secret management in K8s doesn't have a perfect solution yet and I'm just happy that the community is actively working towards creating one(s)!

**Comment by xakraz** [comment]
Hi @jcassee and@devstein ,

Thanks for your quick answers 🙏

I am a beginner in k8s ecosystem and was not really aware of the differences between the plugins (exec vs go).

I asked my self the question when I saw both support for ArgoCD, which was my focus.

I agree with you and like you I am happy to see the community active by creating such nice projects and tools. My goal behind was to see if it would be relevant to join the effort to build a "stronger" tool to become the "default" while working with `sops` :)

**Comment by jcassee** [comment]
An interesting difference between viaduct-ai/kustomize-sops and goabout/kustomize-sopssecretgenerator is that this project basically tries to restore the old `secretGenerator` functionality of `command: sops -d secret.sops.yaml`, while kops uses sops to encrypt whole Kubernetes resource definitions.

**Comment by jcassee** [comment]
I updated the documentation in 23e22e3f994f8cb0d1190b5d5d0cea8c81715cb3. I hope I described this plugin a bit better without pigeon-holing others.

I'm going to close this issue now. I'm not sure what the best channel for talking about these things is. Maybe the [#kustomize Slack channel](https://kubernetes.slack.com/archives/C9A5ALABG)? I'm absolutely fine with doing a "after hours" chat in a closed GitHub issue. :smile:
